### ISSUE TITLE
Tests needed to confirm NN layers function with different optimizers

### ISSUE THREAD

**Issue by nyghtowl** [deeplearning4j/deeplearning4j#617]
Check optimizers, backprop and pretrain/finetune for each layer and setup gradient checks and hyperparameter tests as needed:

1st release
- AutoEncoder
- RBM
- CNN
- GravesLSTM
- Dense
- OutputLayer

next release after...
- GRU
- LSTM Structured
- RNTN

**Comment by nyghtowl** [comment]
DenseLayer needs computeGradientAndScore for pretrain

**Comment by agibsonccc** [comment]
This will also be a no op. DenseLayers are only used for backprop. If you
want to use pretrain with the same conf use an autoencoder.
ᐧ

**Comment by agibsonccc** [comment]
This has been done afaik.
